odl/contrib/fom/supervised.py

```diff
@@ -396,71 +437,60 @@ def blurring(data, ground_truth, mask=None, normalized=False,
-        mean squared error.
+def false_structures_mask(foreground, smoothness_factor=None):
+    """Return mask emphasizing areas outside ``foreground``.
 
     Parameters
     ----------
-    data : `Tensor`
-        Input data to compare to the ground truth.
-    ground_truth : `Tensor`
-        Reference to compare ``data`` to.
-    mask : `array-like`, optional
-        Binary mask to define ROI in which FOM evaluation is performed.
-    normalized  : bool, optional
-        Boolean flag to switch between unormalized and normalized FOM.
+    foreground : `FnBaseVector`
+        The region that should be de-emphasized.
     smoothness_factor : float, optional
-        Positive real number. Higher value gives smoother weighting.
+        Positive real number. Higher value gives smoother transition
+        between foreground and its complement.
 
     Returns
     -------
-    fs : float
-        FOM value, where a lower value means a better match.
-
-    Notes
-    -----
-    The FOM evaluates
-
-    .. math::
-        \mathrm{FS} = \\big \| \\alpha^{-1} (f - g) \\big \|^2_2,
+    result : `FnBaseVector`
+         Euclidean distances of elements in ``foreground``.
 
-    or, in normalized form
-
-    .. math::
-        \mathrm{FS_N} = \\frac{\\big \| \\alpha^{-1}(f - g)\\big \|^2_2}
-                              {\\big \| \\alpha^{-1} f \\big \|^2_2 +
-                               \\big \| \\alpha^{-1} g \\big \|^2_2}.
+    Examples
+    --------
+    >>> space = odl.uniform_discr(0, 1, 5)
+    >>> foreground = space.element([0, 0, 1.0, 0, 0])
+    >>> mask = false_structures_mask(foreground)
+    >>> np.asarray(mask)
+    array([ 0.4,  0.2,  0. ,  0.2,  0.4])
 
-    The weighting function :math:`\\alpha` is given as
+    Raises
+    ------
+    ValueError
+        If foreground is all zero or all one, or contains values not in {0, 1}.
 
-    .. math::
-        \\alpha(x) = e^{-\\frac{1}{k} \\beta_m(x)},
+    Notes
+    -----
+    This helper function computes the Euclidean distance transform from each
+    point in ``foreground.space`` to ``foreground``.
 
-    where :math:`\\beta_m(x)` is the Euclidian distance transform of a
-    given binary mask :math:`m`, and :math:`k` positive real number that
-    controls the smoothness of the weighting function :math:`\\alpha`.
-    The weighting gives higher values to structures outside the region
-    of interest defined by the mask.
+    The weighting gives higher values to structures outside the foreground
+    as defined by the mask.
     """
     from scipy.ndimage.morphology import distance_transform_edt
+    space = foreground.space
 
-    if smoothness_factor is None:
-        smoothness_factor = np.mean(data.space.shape) / 10
+    unique = np.unique(foreground)
+    if not np.array_equiv(unique, [0., 1.]):
+        raise ValueError('`foreground` is not a binary mask or has '
+                         'either only true or only false values {!r}'
```

Comment:
Why is this a problem? (all true/false)

Reply:
I guess it does not make sense, from some perspective? If everything is in the foreground, then no mask is needed. If everything is background, we have nothing to compare to. But no, technically it might not be ok.

Reply:
If it is technically OK we should allow it, else you get irritating bugs. e.g. what happens if you loop over a list of `foregrounds=[nothing, some stuff, everything]`, currently that would (irritatingly) give an error

Reply:
Agreed, even if for no other reason but performance (we do `np.unique` just for the check)